theatre/studio/src/panels/DetailPanel/ProjectDetails.tsx

```diff
@@ -57,35 +58,33 @@ const ProjectDetails: React.FC<{
 
   const exportProject = useCallback(async () => {
     // get all possible asset ids referenced by either static props or keyframes
-    const sheets = Object.values(val(project.pointers.historic.sheetsById))
-    const staticValues = sheets
-      .flatMap((sheet) => Object.values(sheet?.staticOverrides.byObject ?? {}))
-      .flatMap((overrides) => Object.values(overrides ?? {}))
-    const keyframeValues = sheets
-      .flatMap((sheet) => Object.values(sheet?.sequence?.tracksByObject ?? {}))
-      .flatMap((tracks) => Object.values(tracks?.trackData ?? {}))
-      .flatMap((track) => track?.keyframes)
-      .map((keyframe) => keyframe?.value)
-
-    const allValues = [...staticValues, ...keyframeValues].filter(
-      // value is string, and is unique
-      (value, index, self) =>
-        typeof value === 'string' && self.indexOf(value) === index,
-    ) as string[]
 
-    const blobs = new Map<string, Blob>()
+    const allValues = getAllPossibleAssetIDs(project)
 
-    // only export assets that are referenced by the project
-    await Promise.all(
-      allValues.map(async (value) => {
-        const assetUrl = project.assetStorage.getAssetUrl(value)
-        const response = await fetch(assetUrl)
+    const blobs = new Map<string, Blob>()
 
-        if (response.ok) {
-          blobs.set(value, await response.blob())
-        }
-      }),
-    )
+    try {
+      // only export assets that are referenced by the project
+      await Promise.all(
+        allValues.map(async (value) => {
+          const assetUrl = project.assetStorage.getAssetUrl(value)
+
+          const response = await fetch(assetUrl)
```

Comment:
this needs a `try{}`

Reply:
It has one, just around the loop, not the statement itself, so we abort early.